File: src/main/stack/java/Exec.java
import java.io.*;
import java.util.Arrays;

public class Exec {
    public static void main(String[] args) {
        try {
            // Definir os métodos a serem testados
            String[] methods = { "pushIndex", "push", "pushLast",
                                  "popIndex", "pop", "popLast",
                                  "peekIndex", "peek", "peekLast" };

            // Lendo o arquivo de entrada
            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
            String line;

            while ((line = reader.readLine()) != null) {
                if (line.isBlank()) continue;

                int[] input = Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();

                for (String method : methods) {
                    long[] results = new long[30];

                    for (int i = 0; i < 30; i++) {    
                        Stack ck = new Stack();

                        // Adicionando os valores na LinkedList
                        for (String num: line.split(" "))
                            ck.push(Integer.parseInt(num));

                        long start = 0;
                        long end = 0;

                        switch (method) {
                            case "pushIndex":
                                start = System.nanoTime();
                                ck.pushLast(10);
                                end = System.nanoTime();
                                break;

                            case "push":
                                start = System.nanoTime();
                                ck.add(10, input.length / 2);
                                end = System.nanoTime();
                                break;

                            case "pushLast":
                                start = System.nanoTime();
                                ck.pushLast(10);
                                end = System.nanoTime();
                                break;

                            case "popIndex":
                                start = System.nanoTime();
                                ck.popIndex(10);
                                end = System.nanoTime();
                                break;

                            case "pop":
                                start = System.nanoTime();
                                ck.pop();
                                end = System.nanoTime();
                                break;

                            case "popLast":
                                start = System.nanoTime();
                                ck.popLast();
                                end = System.nanoTime();
                                break;

                            case "peekIndex":
                                start = System.nanoTime();
                                ck.peekIndex(10);
                                end = System.nanoTime();
                                break;

                            case "peek":
                                start = System.nanoTime();
                                ck.peek();
                                end = System.nanoTime();
                                break;

                            case "peekLast":
                                start = System.nanoTime();
                                ck.peekLast();
                                end = System.nanoTime();
                                break;
                        }

                        results[i] = (end - start);
                    }

                    Arrays.sort(results); // Ordena para buscar a mediana
                    String outputLine = "stack-java " + results[14] + " " + input.length;

                    // Define o arquivo de saída baseado no método
                    File file = new File(method + ".data");
                    boolean isNewFile = !file.exists() || file.length() == 0;

                    try (PrintWriter writer = new PrintWriter(new FileOutputStream(file, true))) {
                        if (isNewFile) {
                            writer.println("estrutura-linguagem tempo tamanho_da_entrada");
                        }
                        writer.println(outputLine);
                    }
                }
            }
        } catch (IOException ioe) {
            ioe.printStackTrace();
        }
    }
}

File: src/main/stack/python/Exec.py
import sys
import time
import os
from linked_list import LinkedList

methods = ["pushIndex", "push", "PushLast", 
           "popIndex", "pop", "popLast", 
           "peekIndex", "peek", "peekLast"]

for line in sys.stdin:
    line = line.strip()
    if not line:
        continue
    
    input_list = list(map(int, line.split()))
    n = len(input_list)
    
    for method in methods:
        results = []
        
        for _ in range(30):
            ll = Stack(60)
            for num in input_list:
                ll.push(num)
            
            start = 0
            end = 0
            
            if method == "pushIndex":
                start = time.perf_counter_ns()
                ll.pushIndex(10)
                end = time.perf_counter_ns()
            elif method == "push":
                start = time.perf_counter_ns()
                ll.push(10, n // 2)
                end = time.perf_counter_ns()
            elif method == "PushLast":
                start = time.perf_counter_ns()
                ll.PushLast(10)
                end = time.perf_counter_ns()
            elif method == "peekIndex":
                start = time.perf_counter_ns()
                ll.peekIndex()
                end = time.perf_counter_ns()
            elif method == "peek":
                start = time.perf_counter_ns()
                ll.peek(n // 2)
                end = time.perf_counter_ns()
            elif method == "peekLast":
                start = time.perf_counter_ns()
                ll.peekLast()
                end = time.perf_counter_ns()
            elif method == "removeIndex":
                start = time.perf_counter_ns()
                ll.removeIndex()
                end = time.perf_counter_ns()
            elif method == "pop":
                start = time.perf_counter_ns()
                ll.pop(n // 2)
                end = time.perf_counter_ns()
            elif method == "removeLast":
                start = time.perf_counter_ns()
                ll.removeLast()
                end = time.perf_counter_ns()
            results.append(end - start)
        
        results = sorted(results)
        output_line = f"linkedlist-python {results[14]} {n}"
        
        file_name = f"{method}.data"
        is_new_file = not os.path.exists(file_name) or os.stat(file_name).st_size == 0
        
        with open(file_name, "a") as file:
            if is_new_file:
                file.write("estrutura-linguagem tempo tamanho_da_entrada\n")
            file.write(output_line + "\n")
